services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/share_activity_pack/shareActivityPackModal.tsx

```diff
@@ -1,43 +1,102 @@
+const ALL_CLASSES = 'All Classes';
 
-export const ShareActivityPackModal = ({ activityPackData, closeModal, singleActivity, unitId }) => {
+interface ShareActivityPackModalPropsInterface {
+  activityPackData: ActivityPack;
+  closeModal: () => void;
+  selectableClassrooms?: string[];
+  selectedClassroomId?: string;
+  selectedClassroomName?: string;
+  singleActivity: ActivityElement;
+  unitId: string;
+}
+
+export const ShareActivityPackModal = ({ activityPackData, closeModal, selectableClassrooms, selectedClassroomId, selectedClassroomName, singleActivity, unitId }: ShareActivityPackModalPropsInterface) => {
   const [showSnackbar, setShowSnackbar] = React.useState<boolean>(false);
   const [selectedClass, setSelectedClass] = React.useState<DropdownObject>({ label: '', value: '' });
   const [classrooms, setClassrooms] = React.useState<Classroom[]>([]);
   const [classroomUnits, setClassroomUnits] = React.useState<ClassroomUnit[]>([]);
   const [link, setLink] = React.useState<string>(classrooms && classrooms.length && getDefaultLink());
 
   React.useEffect(() => {
-    if(!classrooms.length || (!classroomUnits.length && unitId)) {
+    if(!classroomUnits.length && unitId) {
       requestGet(`/teachers/classrooms/classrooms_and_classroom_units_for_activity_share/${unitId}`, (body) => {
-        setClassrooms(body.classrooms && body.classrooms.classrooms_and_their_students)
+        const classrooms = body.classrooms && filterClassrooms(body.classrooms.classrooms_and_their_students)
+        setClassrooms(classrooms)
         setClassroomUnits(body.classroom_units)
       })
     }
-  }, [])
+  }, []);
 
-  function getDefaultLink() {
-    if(classrooms.length !== 1) {
-      return ''
+  React.useEffect(() => {
+    if(classrooms && classrooms.length) {
+      const newLink = getDefaultLink();
+      const filteredClassrooms = filterClassrooms(classrooms);
+      getSelectedClass(filteredClassrooms);
+      newLink && setLink(newLink);
     }
-    const classroomObject = classrooms[0];
-    const classroomUnitObject = classroomUnits[0];
-    const { classroom } = classroomObject;
-    if(!singleActivity) {
-      const { id } = classroom;
-      return `${process.env.DEFAULT_URL}/classrooms/${id}?unit_id=${unitId}`
-    } else {
-      const { id } = classroomUnitObject;
-      return `${process.env.DEFAULT_URL}/classroom_units/${id}/activities/${singleActivity.id}`;
+  }, [classrooms]);
+
+  React.useEffect(() => {
+    if(classrooms && classrooms.length) {
+      const filteredClassrooms = filterClassrooms(classrooms);
+      setClassrooms(filteredClassrooms);
     }
+  }, [selectedClassroomId, selectedClassroomName, selectableClassrooms]);
+
+  function filterClassrooms(classrooms) {
+    if(selectedClassroomName) {
+      return classrooms.filter(classroomObject => classroomObject.classroom.name === selectedClassroomName);
+    } else if(selectedClassroomId === ALL_CLASSES) {
```

Comment:
you don't need the `else` in this conditional and the next since you are always returning.